### PR TITLE
[docs] Add rtdetr docs

### PR DESCRIPTION
**Pull request by AyushExel**
### Summary
:sparkles::page_facing_up::bookmark_tabs:

This pull request adds documentation for the new RT-DETR model, a vision transformer for object detection. It creates a new page `docs/models/rtdetr.md` and updates the `docs/models/index.md` and `mkdocs.yml` files accordingly.

> _New model `RT-DETR`_
> _Vision transformers for detection_
> _Docs updated in spring_

### Walkthrough
*  Add RT-DETR model to the documentation ([link](https://github.com/ultralytics/ultralytics/pull/2538/files?diff=unified&w=0#diff-6e3a72a975c12fa6d872b564c6d5d8ab61654e0739d82d8cb722f0597a30d3c3L16-R16), [link](https://github.com/ultralytics/ultralytics/pull/2538/files?diff=unified&w=0#diff-cbd5883d261a919ff508c661a9a82380fd847f26b8143142ce6d5a0215df0943L1-R36), [link](https://github.com/ultralytics/ultralytics/pull/2538/files?diff=unified&w=0#diff-98d0f806abc9af24e6a7c545d3d77e8f9ad57643e27211d7a7b896113e420ed2L167-R168))
   * Create a new page `docs/models/rtdetr.md` with detailed information about the model, its tasks, modes, and usage ([link](https://github.com/ultralytics/ultralytics/pull/2538/files?diff=unified&w=0#diff-cbd5883d261a919ff508c661a9a82380fd847f26b8143142ce6d5a0215df0943L1-R36))
   * Include the new page in the navigation menu of the documentation in `mkdocs.yml` ([link](https://github.com/ultralytics/ultralytics/pull/2538/files?diff=unified&w=0#diff-98d0f806abc9af24e6a7c545d3d77e8f9ad57643e27211d7a7b896113e420ed2L167-R168))
   * Update the list of available models in `docs/models/index.md` with a brief introduction and a link to the new page ([link](https://github.com/ultralytics/ultralytics/pull/2538/files?diff=unified&w=0#diff-6e3a72a975c12fa6d872b564c6d5d8ab61654e0739d82d8cb722f0597a30d3c3L16-R16))

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Ultralytics documentation now includes RT-DETR, a real-time object detection model.

### 📊 Key Changes
- Added RT-DETR (Realtime Detection Transformer) to the model documentation.
- Provided an overview, key features, and usage instructions for RT-DETR.
- Listed pre-trained RT-DETR models with performance metrics and supported tasks.
- Updated model index and navigation to reflect the addition of RT-DETR.
- Minor rewording from "Vision Transformers" to "Segment Anything Model (SAM)" for clarity.

### 🎯 Purpose & Impact
- 🚀 Introduces a new model, RT-DETR, which offers real-time performance with high accuracy in object detection tasks.
- 🤝 Enhances the model documentation for better user understanding and ease of use with pre-trained models.
- 🔍 Allows users to make informed decisions about which model to use based on performance metrics (AP and FPS) and task suitability.
- ✅ Enables users to easily implement RT-DETR in their projects with practical code examples and API usage.
- 📈 Potential impact includes improved real-time object detection capabilities for users, aiding various AI applications.

### CODE REVIEW

File: docs/models/rtdetr.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# Realtime Detection Transformers
+
+## About
+
+## Supported Tasks
+
+| Model Type          | Pre-trained Weights | Tasks Supported       |
+|---------------------|---------------------|-----------------------|
+| rtdetr large        | `sam_b.pt`          | Object Detection |
```
Comment (Laughing-q) [inline docs/models/rtdetr.md:28]: the weights here should be rt-detr.😂

Comment (AyushExel) [reply docs/models/rtdetr.md:28]: omg

**Comment by glenn-jocher** [comment]
@AyushExel got it, I'll deploy these now!

**Comment by Luo-Z13** [comment]
Hello @Laughing-q , I meet a problem about RT-DETR in https://github.com/ultralytics/ultralytics/issues/2659, 
can you assist me in solving it?

**Pull request merged**